### PR TITLE
Add Sinatra to external test.

### PR DESCRIPTION
**Pull request by simi**
- bump bake-test-external
- add external directory to gitignore

### CODE REVIEW

File: config/external.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 protocol-rack:
   url: https://github.com/socketry/protocol-rack
   command: bundle exec bake test
+sinatra:
+  url: https://github.com/sinatra/sinatra
+  command: bundle exec rake test
```
Comment (dentarg) [inline config/external.yaml:6]: There is also https://github.com/sinatra/sinatra/tree/main/sinatra-contrib and https://github.com/sinatra/sinatra/tree/main/rack-protection – they have their own tests. I don't know if they should be tested here too, just letting you know.

Comment (ioquatix) [reply config/external.yaml:6]: It's fine to add more external tests as long as they can run without a huge amount of extra scaffolding.

**Comment by ioquatix** [comment]
@simi thanks for your great PR, do you mind rebasing it? Once it's green we can merge.

We may need to sort out <https://github.com/rack/rack/pull/2181> too.

File: config/external.yaml (hunk 1 above)
Comment (simi) [reply config/external.yaml:9]: AFAIK `bake-test-external` doesn't support `Gemfile` in subfolder yet.

Comment (simi) [reply config/external.yaml:9]: ok, I'll check

Comment (ioquatix) [reply config/external.yaml:9]: There was a bug, and I've fixed it and released in v0.5.0

**Comment by ioquatix** [comment]
Should be fixed by <https://github.com/rack/rack/pull/2214>.

File: config/external.yaml (hunk 1 above)
Comment (simi) [reply config/external.yaml:16]: I'll open separate PR to onboard those as well.

File: Gemfile [hunk 2]
```diff
@@ -19,6 +19,6 @@ group :test do
   gem "webrick"
 
   unless ENV['CI'] == 'spec'
-    gem 'bake-test-external'
+    gem 'bake-test-external', '~> 0.3.3'
```
Comment (ioquatix) [inline Gemfile:22]:
```suggestion
    gem 'bake-test-external'
```

**Comment by simi** [comment]
looks finally green @ioquatix

**Review by jeremyevans — APPROVED** [review]

**Comment by dentarg** [comment]
I don't think this should be green?

Both https://github.com/sinatra/sinatra/issues/2018 and https://github.com/sinatra/sinatra/issues/2019 (https://github.com/rack/rack/issues/2218) should fail them

**Comment by dentarg** [comment]
[Looks](https://github.com/rack/rack/actions/runs/9581914215/job/26419753804#step:4:4484) like it is the rack/logger issue that isn't picked up as a failure

**Pull request merged**